scripts/gis-material-update/fetch/fetch_data.sh

```diff
@@ -34,22 +34,36 @@ cfg_layer () {
     echo "${data_object}.layer"
 }
 
+cfg_dest_layer () {
+    echo "${data_object}.dest_layer"
+}
+
+cfg_where_query () {
+    echo "${data_object}.where_query"
+}
+
 # download path
 download_dir=$(parse_config "common.download_path")
 
-# if we have progressed this far, object is supported in configuration
-# now it remains to find out how to download files
+# initialize variables (from configuration YAML)
 addr=$(parse_config $(cfg_addr $data_object))
 layer=$(parse_config $(cfg_layer $data_object))
 local_file=${download_dir}/$(parse_config $(cfg_local_file $data_object))
+dest_layer=$(parse_config $(cfg_dest_layer $data_object))
+where_query=$(parse_config $(cfg_where_query $data_object))
 
 case $data_object in
 hsl|osm)
     wget -O "$local_file" "$addr"
     ;;
+# plain WFS fetch
 ylre_katualue|ylre_katuosat|maka_autoliikennemaarat)
     ogr2ogr -f GPKG "$local_file" WFS:"$addr" "$layer"
     ;;
+# WFS fetch with layer renaming and where condition
+hki)
+    ogr2ogr -f GPKG "$local_file" -nln "$dest_layer" -where "$where_query" WFS:"$addr" "$layer"
```

Comment:
Could this be rewritten as:
```
ogr2ogr -f GPKG "$local_file" "$extra_params" WFS:"$addr" "$layer"
```
and configured like
```
hki:
  extra_params: "-nln alue -where kunta='091'"
...
ylre_katualue:
  extra_params: ""
...
```

Then the same case could be used for all ogr2ogr commands.

The motivation here is to reduce the number of configuration variables that are used for only some of the cases.

Reply:
Thought this earlier, but didn't implement it (due to difficulties in bash script end). Solved bash problems, and implemented as suggested.